src/java/org/apache/cassandra/cql3/statements/CreateTableStatement.java

```diff
@@ -345,6 +345,14 @@ public ParsedStatement.Prepared prepare(Types udts) throws RequestValidationExce
                 if (properties.definedOrdering.size() > columnAliases.size())
                     throw new InvalidRequestException("Only clustering key columns can be defined in CLUSTERING ORDER directive");
 
+                for (ColumnIdentifier orderId: properties.definedOrdering.keySet())
+                {
+                    if (!columnAliases.contains(orderId))
+                    {
+                        throw new InvalidRequestException("Only clustering key columns can be defined in CLUSTERING ORDER directive");
```

Comment:
We can improve the user experience by adding the names of the columns to the exception message, similar to what Tomek did in our fork